packages/ember-data/tests/integration/relationships/has-many-test.js

```diff
@@ -450,6 +450,81 @@ test("A hasMany relationship can be reloaded if it was fetched via ids", functio
   });
 });
 
+test("A hasMany relationship can be reloaded even if it failed at the first time", function() {
+  expect(4);
+
+  Post.reopen({
+    comments: DS.hasMany('comment', { async: true })
+  });
+
+  env.adapter.find = function(store, type, id) {
+    return Ember.RSVP.resolve({ id: 1, links: { comments: "/posts/1/comments" } });
+  };
+
+  var loadingCount = -1;
+  env.adapter.findHasMany = function(store, record, link, relationship) {
+    loadingCount++;
+    if (loadingCount % 2 === 0) {
+      return Ember.RSVP.reject();
+    } else {
+      return Ember.RSVP.resolve([
+        { id: 1, body: "FirstUpdated" },
+        { id: 2, body: "Second" }
+      ]);
+    }
+  };
+  run(function() {
+    env.store.find('post', 1).then(async(function(post) {
```

Comment:
do we actually need these asyncs? qunit supports promises, we merely need to return the promise chain. This will also improve error scenarios in the test suite..